app/assets/javascripts/models/beta/projectBoard.js

```diff
@@ -64,3 +66,95 @@ export const toggleColumn = (projectBoard, column, callback) => {
     return callback.onToggle();
   }
 }
+
+// Drag And drop utils
+const calculatePosition = (aboveStory, bellowStory) => {
+  if (bellowStory === undefined) return Number(aboveStory.position) + 1;
+  if (aboveStory === undefined) return Number(bellowStory.position) - 1;
+  return (Number(bellowStory.position) + Number(aboveStory.position)) / 2;
+};
+
+export const getNewPosition = (
+  destinationIndex,
+  sourceIndex,
+  storiesArray,
+  isSameColumn,
+) => {
+  if(isEmpty(storiesArray)){
+    return 1; // if array is empty than set 1 to story position
+  }
+
+  if (!isSameColumn || sourceIndex > destinationIndex) {
+    return calculatePosition(
+      storiesArray[destinationIndex - 1],
+      storiesArray[destinationIndex],
+    );
+  }
+  
+  return calculatePosition(
+    storiesArray[destinationIndex],
+    storiesArray[destinationIndex + 1],
+  );
+};
+
+// reorder the array
+export const moveTask = (
+  sourceArray,
+  destinationArray,
+  sourceIndex,
+  destinationIndex,
+) => {
+  const newSourceArray = sourceArray;
+  const [removed] = newSourceArray.splice(sourceIndex, 1);
+  const newDestinationArray = destinationArray;
+  newDestinationArray.splice(destinationIndex, 0, removed);
+  return [...newDestinationArray];
+};
+
+export const getNewSprints = (newStories, sprints, sprintIndex) =>
+  sprints.map((sprint, index) =>
+    index === sprintIndex ? { ...sprint, stories: newStories } : sprint,
+  );
+
+export const getNewState = (isSameColumn, dropColumn, currentState) => {
+  if(isSameColumn) {
+    return currentState;
+  }
+  return dropColumn === 'chillyBin' ? status.UNSCHEDULED : status.UNSTARTED;
+}
+
+export const getBacklogStories = (sprints, index) => isEmpty(sprints) ? [{stories: []}] : sprints[index].stories;
+
+export const getArray = (column, backlogArray, chillyBinArray, sprintIndex) => {
```

Comment:
Can you change this function name? `getArray` is too generic 🤔

Reply:
Yep, I will change this.